src/PowerShellEditorServices/Services/PowerShellContext/Handlers/IInvokeExtensionCommandHandler.cs

```diff
@@ -10,16 +10,16 @@
 namespace Microsoft.PowerShell.EditorServices.Handlers
 {
     [Serial, Method("powerShell/invokeExtensionCommand")]
-    public interface IInvokeExtensionCommandHandler : IJsonRpcNotificationHandler<InvokeExtensionCommandParams> { }
+    internal interface IInvokeExtensionCommandHandler : IJsonRpcNotificationHandler<InvokeExtensionCommandParams> { }
 
-    public class InvokeExtensionCommandParams : IRequest
+    internal class InvokeExtensionCommandParams : IRequest
     {
         public string Name { get; set; }
 
         public ClientEditorContext Context { get; set; }
     }
 
-    public class ClientEditorContext
+    internal class ClientEditorContext
```

Comment:
Doesn't `EditorContext.GetEditorContext()` do that? It seems like a public type has already been designed around the editor context concept. What's the scenario here?

Reply:
Couple of things:

1. It's an async method forced to be sync.  Not too big of a deal when called from sync code, but from async code it can cause issues
1. Does more than it needs to.  If I already know the current file and just need selection range or cursor position then it's potentially expensive. Especially since a lot of the code around getting the `ScriptFile` has a lot of validation via catching exceptions
1. Cursor position and range are one based with `GetEditorContext` where the LSP (and this request is zero based

If you don't want to tie the request type details that explicitly to a public type (which is understandable), an alternative would be a new API with this shape:

```csharp
// Probably a bad place for it, open to ideas
public partial class EditorObject
{
    public Task<LspEditorContext> GetLspEditorContextAsync();
}

public class LspPosition
{
    public long Line { get; set; }

    public long Character { get; set; }
}

public class LspRange
{
    public LspPosition Start { get; set; }

    public LspPosition End { get; set; }
}

public class LspEditorContext
{
    public string CurrentFileContent { get; set; }

    public string CurrentFileLanguage { get; set; }

    public string CurrentFilePath { get; set; }

    public LspPosition CursorPosition { get; set; }

    public LspRange SelectionRange { get; set; }
}

Reply:
Gotcha, I can work with this